src/lib/DatePicker.svelte

```diff
@@ -203,6 +206,37 @@
     }
     e.preventDefault()
   }
+  /**
+   * Time picker
+   */
+
+  export let timePicker: boolean = false
+  let hourInput: HTMLInputElement
+  let minuteInput: HTMLInputElement
+
+  function setHours(newHour: number) {
+    if (newHour >= 0 && newHour <= 23) {
+      browseDate.setHours(newHour)
+      browse(browseDate)
+    } else {
+      hourInput.value = '00'
+    }
+  }
+
+  function setMinutes(newMinute: number) {
+    if (newMinute >= 0 && newMinute <= 59) {
+      browseDate.setMinutes(newMinute)
+      browse(browseDate)
+    } else {
+      minuteInput.value = '00'
+    }
+  }
+
+  function timePickerKeydown(e: KeyboardEvent) {
+    if (e.key == 'Enter') {
+      e.stopPropagation()
+    }
+  }
```

Comment:
Why prevent `Enter`?

Reply:
There is a fuction that listens for this key  on the parent. I was thinking that the whole calender should stay visible if you press Enter inside the timepicker.
```ts
 function keydown(e: KeyboardEvent) {
    if (e.key === 'Escape' && visible) {
      visible = false
      e.preventDefault()
      // When the date picker is open, we prevent 'Escape' from propagating,
      // so for example a parent modal won't be closed
      e.stopPropagation()
    } else if (e.key === 'Enter') {
      visible = !visible
      e.preventDefault()
    }
  }```

Reply:
Yeah it's meant to close it when you're finished selecting the date. I think it's best to keep the `Enter` handling unchanged